### PR TITLE
Update Databricks Tools Notebooks for Tools v24.06.0

### LINKED ISSUE #380: Update Databricks Notebook for Tools 24.02
Databricks Tools Demo Notebooks should be updated to support latest version of 

## Issues
1. Qualification notebook file is corrupted.
2. Virtual env setup is incorrect.
3. Platform argument is not passed correctly in the P/Q tools.

### PR DESCRIPTION
**Pull request by parthosa**
Fixes #380. This PR fixes the databricks notebook for compatibility with Tools v24.06.0. Would recommend feedback by testing the changes E2E.

### Qualification User Tools Notebook Template:
1. Fix virtual environment setup for user tools notebook. 
   - Using `%pip install spark_rapids_user_tools` directly since the libraries are automatically notebook scoped and we do not need to create an additional virtual environment.
2. Add `Tools Version` argument to use custom tools version. 
3. Show Console Output as well since it contains the Top Candidates view
4. Fix summary section - Currently it shows `rapids_4_spark_qualification_output.csv` but for user tools it should be `qualification_summary.csv`.
5. Include a Download button that downloads a zip file with output files and logs.

## Future Maintenence:
1. Added a line in the README to record the latest version of tools used to test the notebooks.
2. For future, If we want to make newer version of tools as default:
   - Test the notebooks for this newer version.
   - If everything looks good, we should update the version number in this line.
   - Update `TOOLS_VER` variable in each notebook with the new version number.

## Evaluation
1. Tested for Databricks AWS and Azure platforms.
2. Tested for the following file locations:
    - `/dbfs/<eventlog>`
    - Cannot support `s3` and `abfs` locations because `aws` and `az` CLI are not available.

cc: @amahussein

### CODE REVIEW

**Comment by SurajAralihalli** [comment]
> Cannot support s3 and abfs locations because aws and az CLI are not available.
Note User Tools Notebook might still fail since aws and az CLI cmds are not available.

We can include these limitations in the readme.

**Comment by cindyyuanjiang** [comment]
We should consider moving output info to cells after running the qualification tool. It would be nice if the first cell contains a concise summary of what the qual tool does.

**Comment by parthosa** [comment]
Address review feedback and included a download button to download output results as a zip along with log files.

**Comment by nvliyuan** [comment]
Hi @parthosa  , can we target this pr to v2406? This repo follows the same release strategy as the plugin, thx.

**Comment by parthosa** [comment]
@nvliyuan Changed the target to  branch-24.06. Thank you.

**Comment by parthosa** [comment]
Will create separate PR with updated target branch.

**Review by SurajAralihalli — DISMISSED** [review]
LTGM, tested the notebooks on Azure Databricks.

**Comment by cindyyuanjiang** [comment]
LGTM, tested both notebooks. Thanks @parthosa!

**Comment by parthosa** [comment]
Thank you for the reviews. Holding it off till next minor release of tools which handles the error when CSP CLIs are not present (in this case `databricks` and `aws`)

 https://github.com/NVIDIA/spark-rapids-tools/pull/1035

File: tools/databricks/[RAPIDS Accelerator for Apache Spark] Qualification Tool Notebook Template.ipynb [hunk 1]
```diff
@@ -1,8 +1,630 @@
...
...</td><td>45</td><td>true</td><td>32</td><td>Filter:ColumnarToRow</td><td>46:47</td><td>false</td></tr><tr><td>app-20220209224509-0008</td><td>24</td><td>Filter</td><td>null</td><td>2.8</td><td>0</td><td>41</td><td>true</td><td>31</td><td>null</td><td>null</td><td>false</td></tr><tr><td>app-20220209224509-0008</td><td>24</td><td>BroadcastHashJoin</td><td>null</td><td>5.09</td><td>0</td><td>17</td><td>true</td><td>38</td><td>null</td><td>null</td><td>false</td></tr><tr><td>app-20220209224509-0008</td><td>24</td><td>Project</td><td>null</td><td>3.0</td><td>0</td><td>93</td><td>true</td><td>null</td><td>null</td><td>null</td><td>false</td></tr><tr><td>app-20220209224509-0008</td><td>24</td><td>WholeStageCodegen (1)</td><td>WholeStageCodegen (1)</td><td>2.9</td><td>325673</td><td>19</td><td>true</td><td>38</td><td>Project:Filter:ColumnarToRow</td><td>20:21:22</td><td>false</td></tr><tr><td>app-20220209224509-0008</td><td>24</td><td>Expand</td><td>null</td><td>3.0</td><td>0</td><td>7</td><td>true</td><td>40</td><td>null</td><td>null</td><td>false</td></tr><tr><td>app-20220209224509-0008</td><td>24</td><td>Exchange</td><td>null</td><td>4.2</td><td>6179</td><td>11</td><td>true</td><td>38:4
```
Comment (kuhushukla) [inline tools/databricks/[RAPIDS Accelerator for Apache Spark] Qualification Tool Notebook Template.ipynb:17]: Qualification tool

**Comment by parthosa** [comment]
Based on offline discussion with @viadea:
- Added a note with the current limitation of using single event logs for the Qualification notebook.
- We can merge this PR for now. 
- Plan to submit a subsequent PR to update the notebooks to use newer version of tools, which will remove this 

### Result

image

**Comment by kuhushukla** [comment]
@parthosa what related issue with respect to single log (csp cli creds) needs to be fixed first in order for that limitation to resolve? I see that https://github.com/NVIDIA/spark-rapids-tools/pull/1035 is merged.

**Comment by parthosa** [comment]
Yes @kuhushukla, it is merged and will be available to users from next release.

**Comment by kuhushukla** [comment]
> Yes @kuhushukla, it is merged and will be available to users from next release.

In that case using this for multiple logs note would be deprecated even as part of this change right?

**Review by kuhushukla — APPROVED** [review]
lgtm

**Pull request merged**